### PR TITLE
Also allow copying transferables using withCopyFileToContainer

### LINKED ISSUE #3814: Allow to create a container file from a Transferable
```
var contents = Transferable.of(...);
var containerLocation = "/path/to/file";
withCopyFileToContainer(contents, containerLocation)
```

### PR DESCRIPTION
**Pull request by dajudge**
Fixes #3814

### CODE REVIEW

**Comment by kiview** [comment]
Just referencing https://github.com/testcontainers/testcontainers-java/pull/3815 here, so we keep the connection (and maybe we can integrate the test).

File: core/src/main/java/org/testcontainers/containers/GenericContainer.java [hunk 1]
```diff
@@ -1284,15 +1279,97 @@ public SELF withWorkingDirectory(String workDir) {
         return self();
     }
 
+    private interface CopyFileToContainerWrapper<T> {
```
Comment (kiview) [inline core/src/main/java/org/testcontainers/containers/GenericContainer.java:1282]: Can we find an interface name with more meaning for this? One might argue, that `Transferable` could have been a good abstraction already (we can transfer bytes of a file into the container), but of course, this is already used differently in our codebase now.

Maybe `CopyFileToContainerSource` ? If this is reflected in some parameters being renamed from `wrapper` to `source`, this should also help readability.

File: core/src/main/java/org/testcontainers/containers/Container.java [hunk 2]
```diff
@@ -174,6 +175,15 @@ default SELF withFileSystemBind(String hostPath, String containerPath) {
      */
     SELF withCopyFileToContainer(MountableFile mountableFile, String containerPath);
 
+    /**
+     * Set the file to be copied before starting a created container
```
Comment (kiview) [inline core/src/main/java/org/testcontainers/containers/Container.java:179]:
```suggestion
     * Set the Transferable to be copied before starting a created container
```

It is not really a file, or?

**Review by kiview — COMMENTED** [review]
I think keeping the ordering semantics is an improvement over #3814. So let's maybe include an explicit test for `Transferable` as well?

**Comment by dajudge** [comment]
> I think keeping the ordering semantics is an improvement over #3814. So let's maybe include an explicit test for `Transferable` as well?

Yeah there needs to be a test for sure - just wanted to reach some consensus that we're going in that direction before spending the time on it.

**Pull request closed without merging**